### ISSUE TITLE
[Suggestion] Automatic Waypoint Addition on Waystone Activation

### ISSUE THREAD

**Issue by Rodrigo816** [MehVahdJukaar/mapatlases-neoforge#104]
Hello this is a sugestion:

I would love to see integration with waystones mod, when player's activates a waystone should be added to waypoint automatically.

This feature would streamline the process of marking important locations and ensure players can easily navigate back to activated waystones without manually setting waypoints.

https://www.curseforge.com/minecraft/mc-mods/waystones

**Comment by ClockworkCorvus** [comment]
Bumping this idea, this is the ONE thing this mod is missing.

**Comment by MehVahdJukaar** [comment]
Kind of odd because i got another issue open from people annoyed that some blocks are" automatically" added. That one is just due to then pressing b accidentally.ksnt that some thing that one can do here too? Doesn't take much to press a key

**Comment by ClockworkCorvus** [comment]
Most other minimaps have this function either natively or added via addons, just needs a marker icon and the ability for the map to register when a new waystone is activated. It's a small tweak that sounds insignificant but its clearly significant enough for other minimaps (and their addons) to implement it.

**Comment by MehVahdJukaar** [comment]
yes but map atlases is not a normal minimap mod. It works with the vanilla map system which is quite different rom what would usually done for other map mods. Addint a marker should remain just like in vanilla, something where the player has to actively place one. As i've said ive had the opposite of this report (because people would accidentally press something with the atlas) so I'm pretty sure theres no solution that would satisfy both parties anyways. To add that marker one can just press B while looking at it so its something almost immediate too.

**Comment by pinkygrocket** [comment]
A potential solution that could satisfy both parties is having the option there, but have it disabled by default and toggleable via a config option. In that instance, the option is there for packdevs and players looking to tweak their experience, but it's not present out of the box.

**Comment by snewell92** [comment]
Note for anyone else coming here for this feature; as said here and in other places, I'm able to shift+right-click Waystones, beds, and Named Banners to get mostly what I want for my modpack. Works pretty well. Had some weirdness on my LAN server yesterday but nothing a server restart couldn't fix. (idk about this "press B" deal, that doesn't work for me)

I'm not sure this is "not a normal minimap mod"; player expectations and player experience still apply all the same. When playing with mods like Waystones, there is an expectation that enhances the experience. Waystones are either intentionally crafted and placed, or the naturally generated ones are broken and placed elsewhere, or interacted with and serve as an important landmark / anchor / transportation mechanism. There is already a Xaero's integration config, so I'm not exactly sure where the automatic pin threshold stops/ends.

At the end of the day, when I'm exploring and feeling like a cartographer, I think having meaningful, intentional landmarks (waystones) appear somewhat automatically seems fine. I'd note it down if I had a map and was exploring!

We can be immersive without being tedious, in the spirit of vanilla+ even.

A config is a good for packdevs imo, preserving a 'pure vanilla' ootb "it's just vanilla maps, collated heflpully" mentality.

**Comment by MehVahdJukaar** [comment]
So B is the default place pin button. Check your key mappings and see if you have it mapped. Aiming at a block will place it's marker, midair for pin. As for waystones since people keep asking I guess I'll see if I can add something but I really think it's not that needed as using B every no's and then isn't much of an effort

**Comment by snewell92** [comment]
Edit: On 1.19.2 the place pin feature via 'b' isn't there; opened an issue to request back port of feature. ~I'll open a separate issue for B keybind, I'd like to rebind it.~

**Comment by MehVahdJukaar** [comment]
To rebind a key go in keybinds
